File: web/typescript-cucumber-cypress/cypress/e2e/steps/AddCartPageDefinitions.ts
import { Then, When } from "@badeball/cypress-cucumber-preprocessor";
import HomePageActions from "../pages/HomePageActions";
import ProductPageActions from "../pages/ProductPageActions";

const objHomePage = new HomePageActions;
const objProductPage = new ProductPageActions;
let producto = "";

When("El usuario selecciona la categoria {string}", (category: string) => {
    objHomePage.clickSelectCategoria(category);
    cy.wait(2000);
})

When("El usuario selecciona el producto {string}", (product: string) => {
    producto = product;
    objHomePage.clickSelectProducto(product);
    cy.wait(2000);
})

When("El usuario agrega el producto al carrito", () => {
    objProductPage.clickBotonAddCart();
    cy.on('window:alert', (message: string) => {
        expect(message).contains("Product added");
    });
    cy.wait(2000);
})

Then("El usuario valida su producto agregado en el carrito", () => {
    objProductPage.clickBotonCart();
    cy.wait(5000);

    objProductPage.getProductName(producto).then((value: string) => {
        expect(value).to.equal(producto);
    })

})

File: web/java-maven-cucumber-selenium-testng/src/test/java/com/jcst/definitions/AddCartPageDefinitions.java
package com.jcst.definitions;

import com.jcst.actions.HomePageActions;
import com.jcst.actions.ProductPageActions;
import com.jcst.utils.HelperClass;
import com.jcst.utils.HelperUtils;
import io.cucumber.java.en.And;
import io.cucumber.java.en.Then;
import org.testng.Assert;

import java.util.List;

public class AddCartPageDefinitions {

    HomePageActions objHomePage = new HomePageActions();
    ProductPageActions objProductPage = new ProductPageActions();
    String producto = "";

    @And("El usuario selecciona la categoria {string}")
    public void elUsuarioSeleccionaLaCategoria(String category) {
        objHomePage.clickSelectCategoria(category);
    }

    @And("El usuario selecciona el producto {string}")
    public void elUsuarioSeleccionaElProducto(String product) {
        producto = product;
        objHomePage.clickSelectProducto(product);
    }

    @And("El usuario agrega el producto al carrito")
    public void elUsuarioAgregaElProductoAlCarrito() {
        objProductPage.clickBotonAddCart();
        objProductPage.alertaError();
        HelperUtils.sleep(1);
    }

    @Then("El usuario valida su producto agregado en el carrito")
    public void elUsuarioValidaSuProductoAgregadoEnElCarrito() {
        objProductPage.clickBotonCart();

        List<String> productos = objProductPage.getNombreProductos();
        Assert.assertTrue(productos.contains(producto));
    }
}
